music21/roman.py

```diff
@@ -1111,6 +1214,19 @@ def romanNumeralFromChord(
         elif nationalityStart in ('Fr', 'Sw'):
             keyObj = _getKeyFromCache(chordObj.seventh.name.lower())
 
+    if preferSecondaryDominants:
+        if stepRoman != 'V':  # ignore if already "primary" dominant
+            if chordObj.isDominantSeventh() or chordObj.isMajorTriad():
```

Comment:
Just a cosmetic suggestion, but you don't have else branches, reduce indentation with:
```suggestion
    if (
        preferSecondaryDominants:
        and stepRoman != 'V':  # ignore if already "primary" dominant
        and (chordObj.isDominantSeventh() or chordObj.isMajorTriad())
    ):

Reply:
Thanks.